builtin/fstk/ui.lua

```diff
@@ -78,37 +78,48 @@ end
 
 --------------------------------------------------------------------------------
 function ui.update()
-	local formspec = ""
+	local formspec = {}
 
 	-- handle errors
 	if gamedata ~= nil and gamedata.reconnect_requested then
-		formspec = wordwrap_quickhack(gamedata.errormessage or "")
-		formspec = "size[12,5]" ..
-				"label[0.5,0;" .. fgettext("The server has requested a reconnect:") ..
-				"]textlist[0.2,0.8;11.5,3.5;;" .. formspec ..
-				"]button[6,4.6;3,0.5;btn_reconnect_no;" .. fgettext("Main menu") .. "]" ..
-				"button[3,4.6;3,0.5;btn_reconnect_yes;" .. fgettext("Reconnect") .. "]"
+		local error_message = core.formspec_escape(
+				gamedata.errormessage or "<none available>")
+		formspec = {
+			"size[14,8]",
+			"real_coordinates[true]",
+			("textarea[0.5,1.2;13,5;;%s;%s]"):format(
+				fgettext("The server has requested a reconnect:"), error_message),
+			"box[0.5,1.2;13,5;#000]",
+			"button[2,6.6;4,1;btn_reconnect_yes;" .. fgettext("Reconnect") .. "]",
+			"button[8,6.6;4,1;btn_reconnect_no;" .. fgettext("Main menu") .. "]"
+		}
 	elseif gamedata ~= nil and gamedata.errormessage ~= nil then
-		formspec = wordwrap_quickhack(gamedata.errormessage)
+		local error_message = core.formspec_escape(
+				gamedata.errormessage or "<none available>")
+
 		local error_title
 		if string.find(gamedata.errormessage, "ModError") then
-			error_title = fgettext("An error occurred in a Lua script, such as a mod:")
+			error_title = fgettext("An error occurred in a Lua script:")
 		else
 			error_title = fgettext("An error occurred:")
 		end
-		formspec = "size[12,5]" ..
-				"label[0.5,0;" .. error_title ..
-				"]textlist[0.2,0.8;11.5,3.5;;" .. formspec ..
-				"]button[4.5,4.6;3,0.5;btn_error_confirm;" .. fgettext("Ok") .. "]"
+		formspec = {
+			"size[14,8]",
+			"real_coordinates[true]",
+			("textarea[0.5,1.2;13,5;;%s;%s]"):format(
+				error_title, error_message),
+			"box[0.5,1.2;13,5;#000]",
```

Comment:
Could you maybe please first define the box, then the textarea? Currently boxes are drawn manually before all normal elements, but this might change. All normal elements are drawn in the order they are defined.